### PR TITLE
Add cosign to test image dependency

### PR DESCRIPTION
**Pull request by tiffanny29631**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -425,6 +428,16 @@ clean-golangci-lint:
...
 
+.PHONY: clean-cosign
+clean-cosign:
+	@rm -rf $(COSIGN)
+
+.PHONY: install-cosign
+install-cosign: "$(COSIGN)"
+
+"$(COSIGN)": buildenv-dirs
+	GOPATH="$(GO_DIR)" go install github.com/sigstore/cosign/v2/cmd/cosign@$(COSIGN_VERSION)
```
Comment (nan-yu) [inline Makefile:439]: Set CGO_ENABLED=0 for compatibility with containers missing glibc

File: Makefile.e2e [hunk 2]
```diff
@@ -46,7 +46,7 @@ test-e2e: config-sync-manifest-local __install-nomos-local test-e2e-nobuild
 # Run the Go e2e tests without building images/manifests.
 # Useful for modifying test code and rerunning tests without rebuilding images.
 .PHONY: test-e2e-go-nobuild
-test-e2e-nobuild: "$(KUSTOMIZE)" "$(HELM)" "$(CRANE)"
+test-e2e-nobuild: "$(KUSTOMIZE)" "$(HELM)" "$(CRANE)" "$(COSIGN)"
```
Comment (sdowell) [inline Makefile.e2e:49]: If you're choosing to download this at runtime, it doesn't need to also be installed by the Dockerfile into the image

[the remark this answers is not in the document]
Comment (tiffanny29631) [reply Makefile.e2e:49]: Removed

Comment (tiffanny29631) [reply Makefile.e2e:49]: Removed

Comment (tiffanny29631) [reply Makefile.e2e:49]: Seems like it's still needed for the kind presubmit

Comment (sdowell) [reply Makefile.e2e:49]: It can be added to `build-kind-e2e` below to consistently download at runtime

Comment (tiffanny29631) [reply Makefile.e2e:49]: Yeah just did, see if it works

Comment (tiffanny29631) [reply Makefile.e2e:49]: And it did :]

Comment (sdowell) [reply Makefile.e2e:49]: It's still being installed in the Dockerfile, I'd move it to the `build-kind-e2e` target below for consistency.

Comment (tiffanny29631) [reply Makefile.e2e:49]: Done

**Review by nan-yu — APPROVED** [review]
/lgtm

**Pull request merged**